### DISCUSSION TITLE
Use GPIO as digital input sensor

### DISCUSSION THREAD

**Discussion by krishy2** [arendst/Tasmota#17898] [Ideas]
If you want to use an input as a digital sensor, like gate closed or water present you have no other option just configure it as a switch, then detach it from its relay.
So it would be great to implement a sensor as Digital input, or Bool input. That could be accessed from rules and it would not be connected to any relays.

**Comment by barbudor** [comment]
You can configure a switch even on a board that has no Relay. In that case you don't need to detach.
"Switch" is "Digital Input" in Tasmota language

**Comment by sfromis** [comment]
If you want to handle it with rules, you do not need any detaching. Just see your Switch pin as a "digital input", where you get actions when things change. Simplest is to set `SwitchMode 1`, but other SwitchMode values may be easier to deal with, depending on your use case.

Simple test rule: `rule1 on Switch1#state do var1 %value% endon`

**Comment by krishy2** [comment]
I was not able to “query” the status of a switch, nor use it in a rule (I mean the current value). The objective was to send a KNX telegram on every Teleperiod.

**Reply by sfromis** [reply]
For the specific use case of "every TelePeriod", this rule can work: `rule1 on tele-Switch1 do var1 %value% endon`, as the switch state is *already* included in the `.../SENSOR` payload published.

**Reply by krishy2** [reply]
How to send a KNX message after the wifi is connected? I’ve managed to only after boot, but without network. Also when a Knxreq arrives, %value% cannot be used, the value cannot be referenced. Should a variable used with constant update?

**Reply by sfromis** [reply]
You have rule triggers allowing you to handle various point in the flow, including after wifi getting connection.
https://tasmota.github.io/docs/Rules/#examples-of-available-triggers

As I'm not a user of KNX, I cannot tell you details of how you can handle messages unknown to me. If you post that KNXreq message, I *may* be able to suggest something.

**Reply by barbudor** [reply]
> I was not able to “query” the status of a switch

Most generally you don't query such informatin as Tasmota always publish in real time any change. So you just have to track
If you need to get the latest information while you were "away", you can make the SENSOR message retained with "SensorRetain ON"
But beware look at the Time field or the MQTT retained field to understand if the information you are provided is fresh or old.
